File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq as h

def solution(scoville, K):
    answer = 0
    length = len(scoville)
    h.heapify(scoville)
    while(scoville[0] < K):
        if(length == 1):
            answer = -1
            break
        answer += 1
        no_hot_1 = h.heappop(scoville)
        no_hot_2 = h.heappop(scoville)
        mix = no_hot_1 + (no_hot_2 * 2)
        h.heappush(scoville, mix)
        length -= 1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    

    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>((o1, o2) -> Integer.compare(o1, o2));
        for (int elem : scoville) {
            pq.add(elem);
        }
        while (true) {
            if (pq.peek() >= K) {
                break;
            }
            if (pq.size() == 1) {
                answer = -1;
                break;
            }
            pq.add(pq.poll() + pq.poll() * 2);
            answer += 1;
        }
        return answer;
    }
}